### ISSUE TITLE
Suggestion: Rename / Extend Condition Trait Methods.

### ISSUE THREAD

**Issue by CupOfTeaJay** [bevyengine/bevy#13629]
# Current State

At the moment, the `Condition` trait provides the following methods:
- `and_then`
- `or_else`

These methods are effectively analogs to the boolean `and` and `or` operators. For example, the Update scheduled systems `zoom_camera` and `translate_camera` in the following code snippet will only run if in `GameState::OpponentTurn` **OR** `GameState::PlayerTurn`:

```c
        // Add Update scheduled systems to the main application.
        app.add_systems(
            Update,
            (zoom_camera, translate_camera)
                .run_if(in_state(AppState::InGame))
                .run_if(in_state(AssetsState::Loaded))
                .run_if(in_state(BootState::NotInBoot))
                .run_if(in_state(GameState::OpponentTurn).or_else(in_state(GameState::PlayerTurn))),
        );
```

The readability can sometimes be weird (in my opinion) with this naming convention:

```c
app.add_systems(
    // `my_system` will only get run if the resource `A` exists & resource 'B' exists.
    my_system.run_if(resource_exists::<A>.and_then(resource_exists::<B>)),
);
```

# Proposed Change

Definitely not a necessary change, but I personally would be much happier renaming `and_then` and `or_else` to:
- `and_if`
- `or_if`

Previous examples with updated naming convention:

```c
        // Add Update scheduled systems to the main application.
        app.add_systems(
            Update,
            (zoom_camera, translate_camera)
                .run_if(in_state(AppState::InGame))
                .run_if(in_state(AssetsState::Loaded))
                .run_if(in_state(BootState::NotInBoot))
                .run_if(in_state(GameState::OpponentTurn).or_if(in_state(GameState::PlayerTurn))),
        );
```

```c
app.add_systems(
    // `my_system` will only get run if the resource `A` exists & resource 'B' exists.
    my_system.run_if(resource_exists::<A>.and_if(resource_exists::<B>)),
);
```

# Additional Discussion

As I'm writing this, I think it's also worth noting that Bevy's `not()` function that operates on conditions is also not included as a method for the `Condition` trait (It exists, but not as a member of the trait). For consistency and completeness, would it make sense to bundle all boolean operator analogs under this trait?
- `and_if`
- `nand_if`
- `or_if`
- `nor_if`
- `xor_if`
- `xnor_if`
- `not`

**Comment by BD103** [comment]
Is there a specific reason you opted for the `_if` suffix? Why not does rename the methods to `and` and `or`?

Furthermore, I'm not sure I agree with `not` being added as a method. Usually I prefer to ready `not` before the function, same as using `!` on a boolean. For example, which one reads nicer?

```rust
// Current system
my_system.run_if(not(resource_exists::<A>))

// Proposed system
my_system.run_if(resource_exists::<A>.not())
```

I prefer the first because I won't accidentally miss `not`, which may not be the case with the second.

Overall, though, I like the idea behind these suggestions. `Condition` could definitely use a bit more love. :)

**Comment by CupOfTeaJay** [comment]
No good reason for the suffix. I guess my brain thought that if the suffixes `_then` and `_else` were being dropped another suffix ought to take their place, but I actually prefer your suggestion of just dropping them completely if that's possible / reasonable.

I also agree with your point that the current implementation of `not` is the most readable, and honestly I would prefer using it the way it is too. My reasoning for suggesting otherwise was that if the full suite of logical operators were to be added for conditions, they ought to be under a single common interface (under the `Condition` trait).

But now that I think about it, the fact that `not` is the only unary operator might be justification enough to treat him as separate and leave as is.

Thanks for your input!

**Comment by BD103** [comment]
Awesome! We can make sure to add a link to `not` from `Condition` so others can find it.

**Comment by CupOfTeaJay** [comment]
> Awesome! We can make sure to add a link to `not` from `Condition` so others can find it.

Fantastic - may I be assigned to this? I would love to take a crack at it sometime this week.

**Comment by BD103** [comment]
We don't really do official assignments, but feel free to open a pull request and we'll happily review it!

**Comment by janhohenheim** [comment]
Closing as implemented in #13784
